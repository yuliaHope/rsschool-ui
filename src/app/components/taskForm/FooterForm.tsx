import * as React from 'react';
import { Form, FormGroup, Button } from 'reactstrap';
import { requiredFieldError, requiredFieldSuccess, githubUrlFieldError } from 'core/validation';
import ReduxFormInput from 'components/ReduxFormInput';
import { InjectedFormProps, reduxForm, Field } from 'redux-form';

type Props = {
    submitApi: any;
    taskId: number;
    courseId: string;
};

export type FormData = {
    taskRepo: string;
    studentComment: string;
};

class FooterForm extends React.Component<Props & InjectedFormProps<FormData, Props>, any> {
    constructor(props: any) {
        super(props);
        this.handleFormSubmit = this.handleFormSubmit.bind(this);
    }

    handleFormSubmit = (event: any) => {
        const { submitApi, taskId, courseId } = this.props;
        const { assignmentRepo, studentComment } = event;
        submitApi({
            courseId,
            taskId,
            assignmentRepo,
            studentComment,
        });
    };

    handleChange = (event: any) => {
        this.setState({ [event.target.name]: event.target.value });
    };

    render() {
        const { handleSubmit } = this.props;
        return (
            <small className="text-muted">
                <Form onSubmit={handleSubmit(this.handleFormSubmit)}>
                    <FormGroup>
                        <Field
                            name="assignmentRepo"
                            label="Choose repo"
                            placeholder="Enter link"
                            component={ReduxFormInput}
                            required={true}
                            type="text"
                            validate={[requiredFieldError, githubUrlFieldError]}
                            warn={requiredFieldSuccess}
                            className="form-control form-control-sm"
                        />
                    </FormGroup>
                    <FormGroup>
                        <Field
                            name="studentComment"
                            label="Comments"
                            placeholder="Write comments here"
                            component={ReduxFormInput}
                            required={false}
                            type="textarea"
                            className="form-control form-control-sm"
                        />
                    </FormGroup>
                    <Button type="submit" color="primary" size="sm">
                        Submit
                    </Button>
                </Form>
            </small>
        );
    }
}

export default reduxForm<FormData, Props>({
    form: 'footerForm',
    enableReinitialize: true,
})(FooterForm);
