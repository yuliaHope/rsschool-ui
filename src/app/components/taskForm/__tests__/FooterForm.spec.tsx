import { shallow } from 'enzyme';
import * as React from 'react';
import FooterForm from '../FooterForm';

describe('FooterForm', () => {
    it('renders correctly if no data', () => {
        const output = shallow(<FooterForm taskId={0} courseId={''} submitApi={jest.fn()} />);
        expect(output).toMatchSnapshot();
    });
});
