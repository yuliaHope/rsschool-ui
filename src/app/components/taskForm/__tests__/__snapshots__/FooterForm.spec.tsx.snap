// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterForm renders correctly if no data 1`] = `
<Component
  courseId=""
  destroyOnUnmount={true}
  enableReinitialize={true}
  forceUnregisterOnUnmount={false}
  form="footerForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitApi={[MockFunction]}
  taskId={0}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
